private/buf/bufwire/module_config_reader.go

```diff
@@ -205,6 +216,99 @@ func (m *moduleConfigReader) getModuleModuleConfig(
 	return newModuleConfig(module, config, nil /* Workspaces aren't supported for ModuleRefs */), nil
 }
 
+func (m *moduleConfigReader) getProtoFileModuleSourceConfigs(
+	ctx context.Context,
+	container app.EnvStdinContainer,
+	sourceRef buffetch.SourceRef,
+	workspaceBuilder bufwork.WorkspaceBuilder,
+	configOverride string,
+	externalDirOrFilePaths []string,
+	externalDirOrFilePathsAllowNotExist bool,
+) (_ []ModuleConfig, retErr error) {
+	readBucketCloser, err := m.fetchReader.GetSourceBucket(ctx, container, sourceRef)
+	if err != nil {
+		return nil, err
+	}
+	workspaceConfigs := map[string]struct{}{}
+	for _, workspaceConfigFile := range bufwork.AllConfigFilePaths {
+		workspaceConfigs[workspaceConfigFile] = struct{}{}
+	}
+	moduleConfigs := map[string]struct{}{}
+	for _, moduleConfigFile := range bufconfig.AllConfigFilePaths {
+		moduleConfigs[moduleConfigFile] = struct{}{}
+	}
+	terminateFileProvider := readBucketCloser.TerminateFileProvider()
+	workspaceConfigDirectory := ""
+	moduleConfigDirectory := ""
+	for _, terminateFile := range terminateFileProvider.GetTerminateFiles() {
+		if terminateFile != nil {
+			if _, ok := workspaceConfigs[terminateFile.Name()]; ok {
+				workspaceConfigDirectory = normalpath.Unnormalize(terminateFile.Path())
+				continue
+			}
+			if _, ok := moduleConfigs[terminateFile.Name()]; ok {
+				moduleConfigDirectory = normalpath.Unnormalize(terminateFile.Path())
+			}
+		}
+	}
+	// If a workspace and module are both found, then we need to check of the module is within
+	// the workspace. If it is, we use the workspace. Otherwise, we use the module.
+	moduleInWorkspace := false
+	if workspaceConfigDirectory != "" && moduleConfigDirectory != "" {
+		workspaceConfig, err := bufwork.GetConfigForBucket(ctx, readBucketCloser, readBucketCloser.RelativeRootPath())
+		if err != nil {
+			return nil, err
+		}
+		for _, dir := range workspaceConfig.Directories {
+			workspaceDirectory := filepath.Join(workspaceConfigDirectory, dir)
+			relativePath, err := filepath.Rel(workspaceDirectory, moduleConfigDirectory)
+			if err != nil || relativePath == "." || filepath.IsAbs(relativePath) {
```

Comment:
It looks like we deem that the module is contained if the workspace if we _cannot_ determine a relative path between the two (i.e. the `err != nil` case)? Also, why do we care if `filepath.IsAbs`? 

Could we use some form of the `workspaceDirectoryEqualsOrContainsSubDirPath` helper that already exists in this package?

Reply:
Yeah, I think I should split this out into a helper because this is a bit confusing. The absolute path check is based on the case where the relative path is `../module/is/outside/workspace/dir`, but yeah, that is probably not a great heuristic. I'll find a better way to capture this in a helper.

Reply:
`workspaceDirectoryEqualsOrContainsSubDirPath` does the trick :)